api/v1beta1/types.go

```diff
@@ -148,6 +148,16 @@ type SubnetSpec struct {
 	// listings. If not set the default behavior is to disable flow logging.
 	// +optional
 	EnableFlowLogs *bool `json:"enableFlowLogs"`
+
+	// Purpose: The purpose of the resource. This field can be either
+	// PRIVATE_RFC_1918 or INTERNAL_HTTPS_LOAD_BALANCER. A subnetwork with
+	// purpose set to INTERNAL_HTTPS_LOAD_BALANCER is a user-created
+	// subnetwork that is reserved for Internal HTTP(S) Load Balancing. If
+	// unspecified, the purpose defaults to PRIVATE_RFC_1918. The
+	// enableFlowLogs field isn't supported with the purpose field set to
+	// INTERNAL_HTTPS_LOAD_BALANCER.
+	// +optional
+	Purpose *string `json:"purpose,omitempty"`
```

Comment:
Could we add kubebuilder validation if there are only two valid values?
```
// +kubebuilder:validation:Enum=PRIVATE_RFC_1918;INTERNAL_HTTPS_LOAD_BALANCER
```

Reply:
The comment was a copy from the google cloud SDK but it seems the comment was not up to date, because it said there were two possible values, then later listed all the possible values.

I updated the comment now and added the kubebuilder validation to only accept the possible values.

Reply:
Thanks for looking into it. Looks good to me.